File: nodejs/cucumber/pages/webform.page.ts
import { expect, type Locator, type Page } from '@playwright/test';

export class WebformPage {

  page: Page;
  input: Locator;
  password: Locator;
  textarea: Locator;
  number: Locator;
  city: Locator;
  color: Locator;
  date: Locator;
  range: Locator;
  file: Locator;
  button: Locator;

  constructor(page: Page) {
    this.page = page;
    this.input = page.locator("[id='my-text-id']");
    this.password = page.locator("[name='my-password']");
    this.textarea = page.locator("[name='my-textarea']");
    this.number = page.locator("//select[@name='my-select']");
    this.city = page.locator("//input[@name='my-datalist']");
    this.color = page.locator("[name='my-colors']");
    this.date = page.locator("[name='my-date']");
    this.range = page.locator("[name='my-range']");
    this.file = page.locator("[name='my-file']");
    this.button = page.getByRole("button", { name: "Submit" });
  }

  async set_input(value: string) {
    await this.input.fill(value);
  }

  async set_password(value: string) {
    await this.password.fill(value);
  }

  async set_textarea(value: string) {
    await this.textarea.fill(value);
  }

  async set_number(value: number) {
    await this.number.selectOption({ index: value });
  }

  async set_city(value: string) {
    await this.city.fill(value);
  }

  async set_color(value: string) {
    await this.color.fill(value);
  }

  async set_date(value: string) {
    await this.date.evaluate((elem, value) => elem.setAttribute("value", value), value);
  }

  async set_file(value: string) {
    await this.file.setInputFiles(value);
  }

  async set_range(value: number) {
    await this.range.fill(value.toString());
  }

  async submit() {
    await this.button.click();
  }

}

File: java/cucumber/src/main/java/pages/WebFormPage.java
package pages;

import java.nio.file.Paths;

import com.microsoft.playwright.Page;
import com.microsoft.playwright.options.AriaRole;
import com.microsoft.playwright.Locator;

public class WebFormPage {

    private Page page;
    private Locator input;
    private Locator password;
    private Locator textarea;
    private Locator number;
    private Locator city;
    private Locator file;
    private Locator color;
    private Locator date;
    private Locator range;
    private Locator button;

    public WebFormPage(Page page) {
        this.page = page;
        this.input = page.locator("[id='my-text-id']");
        this.password = page.locator("[name='my-password']");
        this.textarea = page.locator("[name='my-textarea']");
        this.number = page.locator("//select[@name='my-select']");
        this.city = page.locator("//input[@name='my-datalist']");
        this.color = page.locator("[name='my-colors']");
        this.date = page.locator("[name='my-date']");
        this.range = page.locator("[name='my-range']");
        this.file = page.locator("[name='my-file']");
        this.button = page.getByRole(AriaRole.BUTTON, new Page.GetByRoleOptions().setName("Submit"));
    }

    public void set_input(String value) {
        this.input.fill(value);
    }

    public void set_password(String value) {
        this.password.fill(value);
    }

    public void set_textarea(String value) {
        this.textarea.fill(value);
    }

    public void set_number(int value) {
        this.number.selectOption(String.valueOf(value));
    }

    public void set_city(String value) {
        this.city.fill(value);
    }

    public void set_file(String value) {
        this.file.setInputFiles(Paths.get(value));
    }

    public void set_color(String value) {
        this.color.fill(value);
    }

    public void set_date(String value) {
        this.date.evaluate("(node, val) => { node.value = val; }", value);
    }

    public void set_range(int value) {
        this.range.fill(String.valueOf(value));
    }

    public void submit() {
        this.button.click();
    }

}
